api/v1alpha1/project_config_types.go

```diff
@@ -46,6 +86,29 @@ type PromotionPolicy struct {
 	AutoPromotionEnabled bool `json:"autoPromotionEnabled,omitempty" protobuf:"varint,2,opt,name=autoPromotionEnabled"`
 }
 
+// WebhookReceiverConfig describes the configuration for webhook receivers.
+type WebhookReceiverConfig struct {
+	// GitHub contains the configuration for a webhook receiver that is compatible with
+	// GitHub payloads.
+	GitHub *GitHubWebhookReceiver `json:"github,omitempty" protobuf:"bytes,1,opt,name=github"`
+}
+
+type GitHubWebhookReceiver struct {
+	// Secret contains a reference to a Secret in the same namespace as the ProjectConfig.
+	//
+	// The Secret is expected to contain a `token` key with the secret token configured for
+	// in GitHub for the webhook. For more information about this token, please refer to the
+	// GitHub documentation: https://docs.github.com/en/webhooks/using-webhooks/validating-webhook-deliveries
+	SecretRef corev1.LocalObjectReference `json:"secretRef" protobuf:"bytes,1,opt,name=secretRef"`
+}
+
+// WebhookReceiver describes a path used to receive warehouse events and
+// trigger refreshes.
```

Comment:
I would make this comment less warehouse-specific. It won't be long before we need to start dealing with other events. For example, we'll want to know when a PR is merged so we can refresh an in-progress Promotion that's waiting for that PR to be merged. That's all out of scope right now, but if we do this right, a single WebhookReceiver can be used for multiple event types.